### PR TITLE
lib/deprecated: alias mapAttrsFlatten to mapAttrsToList

### PR DESCRIPTION
**Pull request by tie**
## Description of changes

These functions have identical implementation except for argument names.

This PR also updates usage in Nixpkgs.

__mapAttrsFlatten__: https://github.com/NixOS/nixpkgs/blob/b281753417d9604bc2f9c747bdb826e8f82e04eb/lib/deprecated/misc.nix#L214-L215

__mapAttrsToList__: https://github.com/NixOS/nixpkgs/blob/b281753417d9604bc2f9c747bdb826e8f82e04eb/lib/attrsets.nix#L1093-L1096

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: lib/deprecated/misc.nix
Comment (tie) [inline lib/deprecated/misc.nix:216]: Should we also add `warnIf` to formally depreciate `mapAttrsFlatten` in 24.11?

File: nixos/modules/services/networking/openvpn.nix
Comment (philiptaron) [inline nixos/modules/services/networking/openvpn.nix:226]: Pulled in because of a top-level `with` statement 😿

**Review by philiptaron — APPROVED** [review]
This looks right to me, given https://noogle.dev/f/lib/mapAttrsToList. I take no position on the warnIf.

File: nixos/modules/services/networking/openvpn.nix
Comment (tie) [reply nixos/modules/services/networking/openvpn.nix:226]: Welp 🤷
See also https://github.com/NixOS/nixpkgs/issues/208242

Comment (philiptaron) [reply nixos/modules/services/networking/openvpn.nix:226]: See also who it's assigned to. 😉

**Comment by tie** [comment]
@philiptaron, are we waiting for a review from @infinisil (ping :]) before merging?

**Review by infinisil — COMMENTED** [review]
I think a `lib.warn` would be best! (We'd only need a `lib.warnIf (isInOldestRelease 2411)` if the replacement were introduced in the same change)

**Comment by tie** [comment]
@infinisil, done. I’ve used `isInOldestRelease 2405` since we probably want to start printing warnings for NixOS 24.11 release (and unstable release channel) and hopefully remove this function in future releases to avoid confusion.

**Comment by infinisil** [comment]
Ah no we really just need an unconditional `lib.warn` here. The `isInOldestRelease` thing is a bit confusing, but it's not needed here, see also https://github.com/NixOS/nixpkgs/issues/291939

**Comment by infinisil** [comment]
Specifically `isInOldestRelease` is asking "is the new feature that replaces the old one in the oldest supported release?", which is always `true` if the replacement already exists in all supported releases

**Comment by tie** [comment]
Ah, sorry, misunderstood your suggestion.

**Comment by tie** [comment]
@infinisil, changed to `warn` about deprecation instead of `warnIf`.

**Pull request merged**
